public/components/common/apm_toggle.tsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useEffect } from 'react';
+import { EuiSwitch, EuiSpacer, EuiToolTip } from '@elastic/eui';
+import { observabilityApmServicesID } from '../../../common/constants/shared';
+import { coreRefs } from '../../framework/core_refs';
+
+export const ApmToggle = () => {
+  const [isEnabled, setIsEnabled] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    // Read current value from uiSettingsClient
+    const loadSetting = async () => {
+      if (coreRefs.core?.uiSettings) {
+        const apmEnabled = coreRefs.core.uiSettings.get<boolean>('observability:apmEnabled', false);
+        setIsEnabled(apmEnabled);
+      }
+    };
+
+    loadSetting();
+
+    // Subscribe to changes
+    const subscription = coreRefs.core?.uiSettings
+      .get$('observability:apmEnabled')
```

Comment:
Not quite sure if the observable works for workspace level settings. Please test it out.